packages/snap-preact/src/setupEvents.ts

```diff
@@ -48,5 +34,59 @@ export const setupEvents = () => {
 		await next();
 	});
 
+	eventManager.on('controller/recommendation/update', async (data: ControllerRecommendationUpdateData, next: Next) => {
+		const { controllerIds } = data || {};
+
+		//filter through all recommendation controllers for matches with profileIds and realtime config
+		const controllers = matchControllers(controllerIds, (controller) => {
+			return Boolean(controller.type === 'recommendation' && controller.config?.realtime);
+		});
+
+		controllers.map((controller) => {
+			controller.search();
+		});
+
+		await next();
+	});
+
 	return eventManager;
 };
+
+const matchControllers = (
+	matchIds: (string | RegExp)[] | undefined,
+	conditionCallback?: (ctrl: AbstractController) => boolean
+): AbstractController[] => {
+	return Object.keys(window.searchspring.controller || {}).reduce((arr, id) => {
+		const controller = window.searchspring.controller[id] as AbstractController;
+
+		if (conditionCallback && !conditionCallback(controller)) {
+			return arr;
+		}
```

Comment:
Maybe just use a filter on the return of `matchControllers` instead of callback?